__tests__/topics.test.js

```diff
@@ -0,0 +1,121 @@
+const request = require('supertest');
+const db = require('../db/connection.js');
+const app = require('../app.js');
+
+describe('/api/topics', () => {
+  describe('we have to get all topics with slug and description', () => {
+    test('200: status code and contain the expected data type and fields', async () => {
+      const res = await request(app).get('/api/topics');
+      expect(res.statusCode).toBe(200);
+      expect(Array.isArray(res.body)).toBe(true);
+      res.body.forEach(topic => {
+        expect(topic).toHaveProperty('slug');
+        expect(typeof topic.slug).toBe('string');
+
+        expect(topic).toHaveProperty('description');
+        expect(typeof topic.description).toBe('string');
+      });
+    });
+  });
+});
+
+describe('/api/articles:/article_id', () => {
+  describe('GET /api/articles/:article_id', () => {
+    test('200: status code and contain the expected data type and fields', async () => {
+      const res = await request(app).get(`/api/articles/33`);
+      expect(res.statusCode).toBe(200);
+      expect(Array.isArray(res.body)).toBe(true);
+      res.body.forEach(article => {
+        expect(article).toEqual(
+          expect.objectContaining({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String), 
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+          })
+        );
+      });
+    });
+    test('400: when passing wrong type of id', async () => {
+      const res = await request(app).get('/api/articles/asdasd');
+      expect(res.statusCode).toBe(400);
+    });
+
+    test('404: when passing a non-existent id', async () => {
+      const res = await request(app).get(`/api/articles/44444`);
+      expect(res.statusCode).toBe(404);
+    });
+  });
+});
+
+
+
+describe('/api/articles', () => {
+  describe('GET /api/articles', () => {
+    test('200: status code and contain the expected data type and fields', async () => {
+      const res = await request(app).get(`/api/articles`);
+      expect(res.statusCode).toBe(200);
+      expect(Array.isArray(res.body)).toBe(true);
```

Comment:
Please check the length of the array before the forEach() to make sure it's not an empty array. In case it's empty, the .forEach() won't run and we get a false positive test.